src/cpp/statistics/rtps/monitor-service/MonitorServiceListener.cpp

```diff
@@ -86,6 +87,109 @@ void MonitorServiceListener::on_writer_change_received_by_all(
     static_cast<void>(change);
 }
 
+bool MonitorServiceListener::on_incompatible_qos_matching(
+        const fastdds::rtps::GUID_t& local_guid,
+        const fastdds::rtps::GUID_t& remote_guid,
+        const fastdds::dds::PolicyMask& incompatible_qos_policies) const
+{
+    bool ret = true;
+
+    // Convert the PolicyMask to a vector of policy ids
+    std::vector<uint32_t> incompatible_policies;
+    for (uint32_t id = 1; id < dds::NEXT_QOS_POLICY_ID; ++id)
+    {
+        if (incompatible_qos_policies.test(id))
+        {
+            incompatible_policies.push_back(id);
+        }
+    }
+
+    if (!incompatible_policies.empty())
```

Comment:
I think we should take some action when `incompatible_policies` is empty. Probably removing `remote_guid` from the entry for `local_guid` in `extended_incompatible_qos_collection_`